### DISCUSSION TITLE
Use rPi to get BLE JKBMS-data and output to modbus?

### DISCUSSION THREAD

**Discussion by axyaxyfr** [jblance/mpp-solar#311] [General]
I'm using the JKBMS command to get info from my batteries and output it through mqtt to my home assistant. This works like a charm.
My batteries don't have any other means of communication, so this is the only way to get the data from the batteries. The problem is, that my inverter (Deye) doesn't know anything about the BMS and SOC. For now it operates just using voltages, but I would like to have the SOC sent to the inverter too.
Would it be possible to have the rPi output the data to a serial port and send them through modbus to the inverter? As far as I can see, the inverter actually supports JKBMS, but any BMS could potentially be emulated on the output.
Is this possible?

Tia

**Comment by jblance** [comment]
It is possible with some code additions / changes
This area is something I have 'scheduled' for milestone release 0.19

**Reply by jblance** [reply]
Would also need to know how to emulate 'battery talk' to the inverter - then you could query bms and output to inverter (via serial)

**Comment by meccip** [comment]
Hello.
Did you find an answer to send SOC to inverter? I am interested in this. I have data from jk bms by mqtt with esp01 and mpp-solar on raspberry. Processing data into nodered.
